apps/contract-sync-job/src/db.ts

```diff
@@ -63,11 +79,71 @@ export async function getContribution(
   }
 }
 
-export const createJobRun = async (
-  govrn: GovrnProtocol,
-  job: { startDate: Date; completedDate: Date; name: string },
+const getIdOfChain = async (chainId: number) => {
+  const chain = await govrn.chain.get({ where: { chain_id: `${chainId}` } });
+  return chain.id;
+};
+
+export const upsertContribution = async (contribution: ContributionData) => {
+  return await govrn.contribution.upsert({
+    where: {
+      chain_id_on_chain_id: {
+        chain_id: await getIdOfChain(contribution.chain_id),
+        on_chain_id: contribution.on_chain_id,
+      },
+    },
+    create: {
+      name: contribution.name,
+      proof: contribution.proof,
+      details: contribution.details,
+      date_of_engagement: contribution.date_of_engagement,
+      user: { connect: { id: contribution.user_id } },
+      activity_type: { connect: { id: contribution.activity_type_id } },
+      status: {
+        connect: { name: 'minted' },
+      },
+      on_chain_id: contribution.on_chain_id,
+      chain: { connect: { chain_id: `${contribution.chain_id}` } },
+    },
+    update: {
+      name: { set: contribution.name },
+      on_chain_id: { set: contribution.on_chain_id },
+      // proof: { set: contribution.proof },
+      // details: { set: contribution.details },
```

Comment:
Why comment these out?

Reply:
I fixed it: 56aca17cb39c0b83874bd426ab8618b83dbb834b

Prisma accepts only `string | null` values, but it crashes when the value is `undefined`.